src/main/java/com/aws/greengrass/logmanager/model/LogFileGroup.java

```diff
@@ -1,36 +1,105 @@
 package com.aws.greengrass.logmanager.model;
 
 import com.aws.greengrass.logmanager.exceptions.InvalidLogGroupException;
-import com.aws.greengrass.util.Utils;
 import lombok.Getter;
 
 import java.io.File;
 import java.net.URI;
 import java.time.Instant;
 import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Comparator;
 import java.util.List;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
 
 
 public final class LogFileGroup {
-    @Getter
     private List<LogFile> logFiles;
     private static Map<String, LogFile> fileHashToLogFile;
     @Getter
     private final Pattern filePattern;
     private final URI directoryURI;
     private final Instant lastUpdated;
 
-    private LogFileGroup(List<LogFile> files, Pattern filePattern, URI directoryURI, Instant lastUpdated) {
-        this.logFiles = files;
+    /**
+     * Creates a log file group without initializing the logFiles. To initialize them you must call
+     * getLogFiles.
+     * @param filePattern - the pattern use to match files on the provided directoryURI
+     * @param directoryURI - the directory where log files are located
+     * @param lastUpdated - instant used to filter out log files last modified after its value
+     */
+    public LogFileGroup(Pattern filePattern, URI directoryURI, Instant lastUpdated) {
+        this.logFiles = new ArrayList<>();
         this.filePattern = filePattern;
         this.directoryURI = directoryURI;
         this.lastUpdated = lastUpdated;
     }
 
+    public List<LogFile> getLogFiles() {
+       return getLogFiles(listDirectory());
+    }
+
+    /**
+     * Gets the log files for the component, by listing all the files in the directory and filtering out the ones
+     * that do not match the log group pattern and have been last modified after the provided lastUpdated value.
+     * @param allDirectoryLogFiles - files in a directory
+     */
+    public List<LogFile> getLogFiles(File... allDirectoryLogFiles) {
```

Comment:
The reason I exposed this tests was so I could better simulate the issue we are trying to address or logs rotating as soon as we scan the directory for logs

Reply:
Making this package private instead of public may help to convey intent